saleor/graphql/discount/mutations/voucher/voucher_update.py

```diff
@@ -84,6 +74,33 @@ def clean_voucher_usage_setting(cls, instance, data):
                         )
                     }
                 )
+        if "usage_limit" in data and instance.usage_limit != data["usage_limit"]:
```

Comment:
What if the user updates the voucher with the same value as before? 
You should add a test case to cover such a case.